shared/src/commonMain/kotlin/org/hyperskill/app/main/presentation/AppReducer.kt

```diff
@@ -165,8 +169,47 @@ class AppReducer(
             state to emptySet()
         }
 
+    private fun handleFirstProblemOnboardingDataFetched(
+        state: State,
+        message: Message.FirstProblemOnboardingDataFetched
+    ): ReducerResult =
+        if (state is State.Ready && state.profile != null && !state.profile.isNewUser) {
```

Comment:
Before current changes `State.Ready.profile` was used to temporarily store the profile between some messages and actions while determining whether should navigate the user to the notification onboarding screen or not, now this exceeded with the same logic about first problem onboarding.

But at some point we should clear `profile` from state before we do this in `navigateUserAfterNotificationOnboarding`, now I suppose we can do this in `handleFirstProblemOnboardingDataFetched` and check for `state.profile != null` in `handleFirstProblemOnboardingCompleted` useless

cc @XanderZhu

Reply:
@ivan-magda yes, I totally agree with you thanks for that unobvious remark! 🤝